### DISCUSSION TITLE
SODA Foundation Starter Guides and Links

### DISCUSSION THREAD

**Discussion by skdwriting** [sodafoundation/soda#24] [General]
Dear All,
In this thread:
- You can find the introduction to SODA Foundation
- SODA Community Channels
- SODA Projects
- How to Start your contribution
- and more
*You can also add useful information under the comments from time to time. It will help the community!*

**Where can I find the SODA Foundation introduction and various activities by SODA Foundation?**
 **[Visit SODA Foundation website](https://www.sodafoundation.io/)**

**How to join as a SODA Foundation Member (Organizations and Institutions)**
 **[Visit Member Joining Page](https://www.sodafoundation.io/join/)**

**I am in individual. I want to contribute. How to join SODA Community?** (Contribution can be code, test, doc and other areas like ecosystem, community building, mentoring, building open source awareness, and more!
[**Join SODA Slack**](http://www.sodafoundation.io/slack/)
After joining the SODA Slack, join to the channel named **#new-contributors**

**Where can I find SODA Projects Source Code?**
**[Github for SODA Foundation Projects](https://github.com/sodafoundation)
[Github for SODA Crystal - The Metadata Engine ](https://github.com/sodafoundation/crystal)
[Github for SODA Container Data Management Projects](https://github.com/soda-cdm)
[Github for SODA Lake Projects](https://github.com/soda-lake)**
Major projects and source code links:
**[Delfin : Heterogeneous Storage Monitoring Project](https://github.com/sodafoundation/delfin)
[Strato: Multicloud Data Management Project](https://github.com/sodafoundation/multi-cloud)
[Kahu: Container Data Protection Project](https://github.com/soda-cdm/kahu)
[Como: Virtual Data Lake Project](https://github.com/soda-lake/como)**

**Where can I find the SODA Foundation Project Documentation?**
Two ways you can get access to the project documentation
a) Visit the respective source code link of the project (as mentioned above)
b) Visit the **[SODA Foundation Documentation Page](https://docs.sodafoundation.io/)**

**What are the other channels through which I can connect to SODA Community?**
**[SODA LinkedIn Page](https://www.linkedin.com/company/sodafoundation)
[SODA Twitter](https://twitter.com/sodafoundation)
[SODA India Meetup](https://www.meetup.com/sodafoundationindia/)**

**Which are the regular meetings I can join to learn and contribute more?**
**[SODA Global Community Meeting](http://bit.ly/sodaglobalcommunitymeeting)
[SODA Design Meeting](http://bit.ly/soda-design-meeting)**

**How to start contributing?**
a)  [**Join SODA Slack**](http://www.sodafoundation.io/slack/)  and say "I want to contribute" in #general channel or #new-contributors channel
b) Go to specific project github and open the Issues tab. There will be different open issues, You can add a comment and say you plan to contribute. The maintainers will reply / assign that issue to you.

**How to connect with maintainers and other developers?**
a) [**Join SODA Slack**](http://www.sodafoundation.io/slack/) and start chatting. You can find different project channels. Respective maintainers will be part of those channels
b) You can add comments to github issues of respective projects. Maintainers will check these regularly and reply to you
c) Join **[SODA Global Community Meeting](http://bit.ly/sodaglobalcommunitymeeting) OR [SODA Design Meeting](http://bit.ly/soda-design-meeting)**

**Where can I find SODA Foundation Project Releases?**
[SODA Releases](https://github.com/sodafoundation/soda/releases)

**What is needed for the development or testing of SODA Projects?**
1. Laptop with Ubuntu (18.04LTS / 20.04LTS) [Please check the project environment for latest changes]
2. Check the project languages - you need an editor for code changes (like vscode) [Currently most of the projects are in go, python, js, ansible etc]. You can get this information from the project source code link (check the above sections to reach the project source code)
3. Internet connection

**Do you have more questions?!**
Ask in [**Join SODA Slack**](http://www.sodafoundation.io/slack/)  or add under this discussion thread.

**Please check the comments section of this discussion thread. I will be adding more tips and info there! :)**
Enjoy open source with SODA Foundation Projects!

**Comment by anvithks** [comment]
This is very helpful @skdwriting.

**Comment by skdwriting** [comment]
### **General Tips and Information : Github**

**[Github Documentation](https://docs.github.com/en)**  is a very good place for learning github. Issue creation, project management, raising PR, and more. You can go to this link and search the keywords to find the information or traverse through the menu!

**Basics of GitHub:**  covering: how to create account, basic operation of creating the repo, branch etc with github. **[Read the basic github operation here](https://docs.github.com/en/get-started/quickstart/hello-world)**

**How to contribute to projects using github?**
**[Read the steps to contribute to a project using github here](https://docs.github.com/en/get-started/quickstart/contributing-to-projects)**

**Reply by skdwriting** [reply]
### **Learning Git with All needed commands. A very good video tutorial:**
[https](https://www.youtube.com/watch?v=SWYqp7iY_Tc)[://www.youtube.com/watch?v=SWYqp7iY_](https://www.youtube.com/watch?v=SWYqp7iY_Tc)[Tc](https://www.youtube.com/watch?v=SWYqp7iY_Tc)

**Comment by skdwriting** [comment]
### **Learn Linux Terminal and Linux Commands? Some useful online terminals, videos, and information are available here**

### **Running Linux Terminal Online: (This helps you to start your hands-on easily from the browser!)**
https://copy.sh/v86/?profile=linux26&ref=its-foss  (Opens terminal, no login needed)
https://linuxcontainers.org/lxd/try-it/    (30 min free linux container)
https://bellard.org/jslinux/?ref=its-foss (login not needed)
https://www.onworks.net/os-distributions/ubuntu-based/free-ubuntu-online-version-20  (Desktop version, ads!)

_More Options: https://itsfoss.com/online-linux-terminals/_

### **Linux Terminal and Commands, Videos:**
60 commands in 10min: https://www.youtube.com/watch?v=gd7BXuUQ91w
https://www.youtube.com/watch?v=iwolPf6kN-k
https://www.youtube.com/watch?v=2PGnYjbYuUo

### Linux Commands Info:
https://www.guru99.com/linux-commands-cheat-sheet.html
https://www.pcwdld.com/linux-commands-cheat-sheet#wbounce-modal

**Comment by skdwriting** [comment]
**Github Commands Cheat Sheet**
![github commands](https://user-images.githubusercontent.com/8660550/230672525-d327f4b4-7523-4aef-9ec5-16003a915063.jpg)
